src/fidesops/service/masking/strategy/masking_strategy_aes_encrypt.py

```diff
@@ -60,3 +93,47 @@ def data_type_supported(data_type: Optional[str]) -> bool:
         """Determines whether or not the given data type is supported by this masking strategy"""
         supported_data_types = {"string"}
         return data_type in supported_data_types
+
+    @staticmethod
+    def _generate_nonce(
+        value: Optional[str],
+        key: str,
+        privacy_request_id: Optional[str],
+        masking_meta: Set[MaskingSecretMeta],
+    ) -> bytes:
+        salt: str = SecretsUtil.get_or_generate_secret(
+            privacy_request_id,
+            [meta for meta in masking_meta if meta.secret_type == SecretType.salt_hmac][
+                0
+            ],
+        )
+        # fixme: replicate what Vault does - remove lower bytes
+        return hmac_encrypt_return_bytes(
+            value, key, salt, HmacMaskingConfiguration.Algorithm.sha_256
+        )
+
+    @staticmethod
```

Comment:
I'm not sure these ought to be staticmethods either

Reply:
Same as above

Reply:
this one, on the other hand, is only called as a static method. Is there a benefit to preferring instance methods instead here?